2 - functional intro, data manipulation/andreea-ex1.js

```diff
@@ -0,0 +1,195 @@
+                    }
+                }
+            }
+        }
+    }
+    return stockAllTrueJAckpot;
+};
+
+/*EX02-Search by type*/
+var gamesFlatenedNames =  flatenedGames(games,'name');
+var gamesFlatenedType =  flatenedGames(games,'type');
+
+function search(dataToSearch,data){
+    if(typeof(dataToSearch) === "boolean"){
+        dataToSearch.toString();
+    }else{
+        dataToSearch
+            .trim()
+            .toLowerCase();
+    }
+    var stock = [];
+    for( var item = 0; item < data.length-1; item++){
+        if(typeof(data[item]) === "boolean"){
+            data[item].toString();
+        }else{
+            data[item]
+                .trim()
+                .toLowerCase();
+        }
+        if(data[item] == dataToSearch){
+            stock.push(dataToSearch);
+        }
+    }
+    $('#message').empty();
+    return stock;
+}
+
+/*Actions and tamplate*/
+$('#ButtonName').on('click',function(){
+    var getNameData = $('#searhByName').val();
+    var flatenedName =  search(getNameData,gamesFlatenedNames);
+    $('#messageName').empty();
+    if(flatenedName[0] == getNameData ){
+        $('#messageName').append("<p class='succes'>Selected answer: " + flatenedName[0].toUpperCase() + "</p>");
+    }else{
+        $('#messageName').append("<p class='error'>Name not found. Please try again!</p>");
+    }
+});
+$('#ButtonType').on('click',function(){
+    var getTypeData = $('#searhByType').val();
+    var flatenedType =  search(getTypeData,gamesFlatenedType);
+    $('#messageType').empty();
+    if(flatenedType[0] == getTypeData ){
+        $('#messageType').append("<p class='succes'>Type of the game is " + flatenedType[0].toUpperCase() + "</p>");
+    }else{
+        $('#messageType').append("<p class='error'>Type not found. Please try again!</p>");
+    }
+});
+
+//Jackpot filter
+    var takeTruJackpots = boolProp(games, true);
+    var takeFalseJackpots = boolProp(games, false);
+
+    $('#searchJackpot').change(function(){
+        if(this.checked == true ){
+            var index;
+            $('#ListJackpots').empty();
+            for(index = 0; index < takeTruJackpots.length; index++ ){
+                $('#ListJackpots').append("<li>" + takeTruJackpots[index] + "</li>")
+            }
+        }else{
+            $('#ListJackpots').empty();
+            for(index = 0; index < takeFalseJackpots.length; index++ ){
+                $('#ListJackpots').append("<li>" + takeFalseJackpots[index] + "</li>")
+            }
+        }
+    });
+
+/*ex2 - Implement a select tag*/
+const peopleArray = [
+    { id: 123, name: 'dave', age: 23 },
+    { id: 456, name: 'chris', age: 23 },
+    { id: 789, name: 'bob', age: 23 },
+    { id: 101, name: 'tom', age: 23 },
+    { id: 102, name: 'tim', age: 23 }
+];
+
+peopleArray.map(
+        function(item) {
+            return $('#selectPeople').append("<option value=" + item.id + ">" + item.id +  "</option>");
```

Comment:
It is better to create a huge html object and append it at the end, then append this on every loop. It is very expensive to search for DOM nodes so it is better to append everything just once.